llvm-external-projects/iree-dialects/lib/Dialect/LinalgExt/Utils/Utils.cpp

```diff
@@ -71,6 +71,27 @@ SmallVector<int64_t> computeInterchangeFromDimPos(ArrayRef<int64_t> dimsPos,
   return interchangeVector;
 }
 
+Value createValueFrom2DConstant(const float *val, int64_t rows, int64_t cols,
+                                bool transpose, Location loc,
+                                PatternRewriter &rewriter) {
+  SmallVector<float> vector(rows * cols, 0.0);
+  for (int i = 0; i < rows; i++) {
+    for (int j = 0; j < cols; j++) {
+      if (!transpose) {
+        vector[i * cols + j] = val[i * cols + j];
+      } else {
+        vector[j * rows + i] = val[i * cols + j];
+      }
+    }
+  }
+  SmallVector<int64_t> shape{rows, cols};
+  if (transpose)
+    shape = {cols, rows};
+  return rewriter.create<arith::ConstantOp>(
```

Comment:
I think you can just create an `ArrayRef` directly from the pointer passed in, and pass the`ArrayRef` into the`::get` method. The copy seems unnecessary.

Reply:
Makes sense. In the case where no transpose is required, then the copy should not be necessary. But when the transpose happens we will need to copy to convert from row major to column major right?

Reply:
Also in the case where there is no transpose, I think we still need to copy since the constant being passed in is defined as a compile time constant (constexpr) and DenseFPElementsAttr::get doesn't expect that.

Reply:
Sure I can remove constexpr and add the transposed values as well. Then we wont need to do this copy.